### PR TITLE
Fix flaky ClientRequestTest

### LINKED ISSUE #27578: Fix flaky ClientRequestTest
### Description

[io.quarkus.it.micrometer.prometheus.ClientRequestTest](https://github.com/quarkusio/quarkus/blob/fac6d3e16939038ec95e330ffb2255e58c12e0dd/integration-tests/micrometer-prometheus/src/test/java/io/quarkus/it/micrometer/prometheus/ClientRequestTest.java) has been disabled because of inconsistent results and it needs to be fixed and enabled.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by brunobat**
fixes #27578

### CODE REVIEW

File: integration-tests/micrometer-prometheus/src/test/java/io/quarkus/it/micrometer/prometheus/Util.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                })
+                .collect(Collectors.joining(","));
+    }
+
+    public static <T> void waitForMeters(Collection<T> collection, int count) throws InterruptedException {
+        int i = 0;
+        do {
+            Thread.sleep(3);
+        } while (collection.size() < count && i++ < 10);
+    }
```
Comment (gsmet) [inline integration-tests/micrometer-prometheus/src/test/java/io/quarkus/it/micrometer/prometheus/Util.java:62]: We usually use Awaitility for that in Quarkus. You should find plenty of examples in the tree.

Comment (brunobat) [reply integration-tests/micrometer-prometheus/src/test/java/io/quarkus/it/micrometer/prometheus/Util.java:62]: yes, will update

**Comment by gsmet** [comment]
The good news is that it's not flaky anymore :].

**Comment by brunobat** [comment]
Yeah... will have to create a specific endpoint on the service to get the test data.

**Comment by gsmet** [comment]
@brunobat please be a bit more conservative with our CI resources:
- make sure your stuff compiles and is well formatted before pushing (at least)
- please don't push any merge commits because we will ask you to rebase anyway to get rid of it before we can merge

Thanks!

**Review by ebullient — APPROVED** [review]

**Comment by geoand** [comment]
So is this ready to go?

**Pull request merged**